### ISSUE TITLE
Memory leak with found with icalrecurtest

### ISSUE THREAD

**Issue by winterz** [libical/libical#801]
```
[1173]$ ./bin/icalrecurtest -f ./bin/icalrecur_test_rscale_withicu_dangi.txt
0 tests failed out of 1.

=================================================================
==3211016==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 6 byte(s) in 1 object(s) allocated from:
    #0 0x7f639ca814b8 in strdup (/lib64/libasan.so.8+0x814b8) (BuildId: a42b37cdd3e851809894c84b99bfdb7c8fa9e357)
    #1 0x7f639d4ec8f8 in icalrecurrencetype_from_string /home/allen/projects/libical/libical/src/libical/icalrecur.c:608

SUMMARY: AddressSanitizer: 6 byte(s) leaked in 1 allocation(s).
```

```
[1174]$ ./bin/icalrecurtest -f ./bin/icalrecur_test_rscale.txt
0 tests failed out of 14.

=================================================================
==3211118==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 138 byte(s) in 14 object(s) allocated from:
    #0 0x7fe56f2814b8 in strdup (/lib64/libasan.so.8+0x814b8) (BuildId: a42b37cdd3e851809894c84b99bfdb7c8fa9e357)
    #1 0x7fe56fcb48f8 in icalrecurrencetype_from_string /home/allen/projects/libical/libical/src/libical/icalrecur.c:608

SUMMARY: AddressSanitizer: 138 byte(s) leaked in 14 allocation(s).
```

**Comment by winterz** [comment]
3.0 branch, btw

**Comment by winterz** [comment]
fixed by #802
